lib/build.js

```diff
@@ -212,54 +216,74 @@ function build (gyp, argv, callback) {
+      }
 
-    if (win) {
-      // Turn off the Microsoft logo on Windows
-      argv.push('/nologo')
-    }
+      if (win) {
+        // Turn off the Microsoft logo on Windows
+        argv.push('/nologo')
+      }
 
-    // Specify the build type, Release by default
-    if (win) {
-      var p = arch === 'x64' ? 'x64' : 'Win32'
-      argv.push('/p:Configuration=' + buildType + ';Platform=' + p)
-      if (jobs) {
-        var j = parseInt(jobs, 10)
-        if (!isNaN(j) && j > 0) {
-          argv.push('/m:' + j)
-        } else if (jobs.toUpperCase() === 'MAX') {
-          argv.push('/m:' + require('os').cpus().length)
+      // Specify the build type, Release by default
+      if (win) {
+        var p = arch === 'x64' ? 'x64' : 'Win32'
+        argv.push('/p:Configuration=' + buildType + ';Platform=' + p)
+        if (jobs) {
+          var j = parseInt(jobs, 10)
+          if (!isNaN(j) && j > 0) {
+            argv.push('/m:' + j)
+          } else if (jobs.toUpperCase() === 'MAX') {
+            argv.push('/m:' + require('os').cpus().length)
+          }
+        }
+      } else {
+        argv.push('BUILDTYPE=' + buildType)
+        // Invoke the Makefile in the 'build' dir.
+        argv.push('-C')
+        argv.push('build')
+        if (jobs) {
+          var j = parseInt(jobs, 10)
+          if (!isNaN(j) && j > 0) {
+            argv.push('--jobs')
+            argv.push(j)
+          } else if (jobs.toUpperCase() === 'MAX') {
+            argv.push('--jobs')
+            argv.push(require('os').cpus().length)
+          }
+        }
+      }
+
+      if (win) {
+        // did the user specify their own .sln file?
+        var hasSln = argv.some(function (arg) {
+          return path.extname(arg) == '.sln'
+        })
+        if (!hasSln) {
+          argv.unshift(gyp.opts.solution || guessedSolution)
         }
       }
     } else {
-      argv.push('BUILDTYPE=' + buildType)
-      // Invoke the Makefile in the 'build' dir.
-      argv.push('-C')
-      argv.push('build')
+      // build with ninja
+      if (verbose) {
+        argv.push('-v')
+      }
+      // Specify the build type, Release by default
+      argv.push('-C', path.join('build', buildType))
       if (jobs) {
-        var j = parseInt(jobs, 10)
+        var j = jobs.toUpperCase() === 'MAX'? require('os').cpus().length : parseInt(jobs, 10)
         if (!isNaN(j) && j > 0) {
-          argv.push('--jobs')
-          argv.push(j)
-        } else if (jobs.toUpperCase() === 'MAX') {
-          argv.push('--jobs')
-          argv.push(require('os').cpus().length)
+          argv.push('-j' + j)
         }
       }
-    }
-
-    if (win) {
-      // did the user specify their own .sln file?
-      var hasSln = argv.some(function (arg) {
-        return path.extname(arg) == '.sln'
-      })
-      if (!hasSln) {
-        argv.unshift(gyp.opts.solution || guessedSolution)
+      // invoke vcvarsall.bat before build
+      if (win && vcDir) {
```

Comment:
Yeah, I have recently committed it: https://github.com/indutny/gyp.js/commit/ce93190063b323abf47f390845d57177424e3cf0 (NOTE: that this is just a starting commit, I have fixed many things since then)